### PR TITLE
feat(argo-cd): Add ability to handle cluster credentials

### PR DESCRIPTION
**Pull request by mkilchhofer**
I thought it would be a good idea to implement _External Cluster Credentials/Clusters_ from [Declarative Setup](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#clusters)

I was not sure, where to put the config in the values. (Also opened an issue here: https://github.com/argoproj/argo-helm/issues/631)

Checklist:

* [X] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [X] I have signed the DCO and the build is green.
* [X] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ Helm v3 has removed the `install-crds` hook so CRDs are now populated by files i
 | global.hostAliases | Mapping between IP and hostnames that will be injected as entries in the pod's hosts files | `[]` |
 | nameOverride | Provide a name in place of `argocd` | `"argocd"` |
 | installCRDs | Install CRDs if you are using Helm2. | `true` |
+| configs.clusterCredentials | [External Cluster Credentials](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#clusters)| `[]` (See [values.yaml](values.yaml)) |
```
Comment (oliverbaehler) [inline charts/argo-cd/README.md:79]: Needs to be more precise. Something like "Provide one or multiple [external cluster credentials](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#clusters)"

**Review by oliverbaehler — CHANGES_REQUESTED** [review]
Improve Values description & Update with Master

**Review by oliverbaehler — APPROVED** [review]
LGTM

**Comment by oliverbaehler** [comment]
@spencergilbert Could we also merge this? Super useful feature imo. @mkilchhofer Please Sync with master

File: charts/argo-cd/templates/argocd-configs/cluster-secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  name: {{ include "argo-cd.name" $ }}-cluster-{{ .name }}
+  labels:
+    {{- include "argo-cd.labels" (dict "context" $) | nindent 4 }}
+    argocd.argoproj.io/secret-type: cluster
+  {{- with .annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+type: Opaque
+data:
```
Comment (spencergilbert) [inline charts/argo-cd/templates/argocd-configs/cluster-secrets.yaml:15]: Since the value file has raw strings for these, shouldn't this be stringData?

Comment (mkilchhofer) [reply charts/argo-cd/templates/argocd-configs/cluster-secrets.yaml]: You are right :) thx for noticing that!
Already pushed a fix.

Comment (spencergilbert) [reply charts/argo-cd/templates/argocd-configs/cluster-secrets.yaml]: Yay, glad I didn't miss something obvious :)

**Review by spencergilbert — APPROVED** [review]
Thanks!

**Pull request merged**